query/query.go

```diff
@@ -116,6 +118,29 @@ func (q *Const) String() string {
 	return "FALSE"
 }
 
+// RepoRegex contains a regex pattern (not compiled) which is matched against
+// repo names. The pattern is lazily compiled by the first indexed searcher that
+// wants to match the pattern against a shard.
+type RepoRegex struct {
+	// any valid regexp pattern
+	Pattern string
+
+	compileOnce sync.Once
```

Comment:
rather do what we do for `type Regex` and make the assumption that it has already been compiled.